seisflows/workflow/inversion.py

```diff
@@ -419,10 +447,14 @@ def _evaluate_line_search_misfit(self):
             [self.run_forward_simulations,
              self.evaluate_objective_function],
             path_model=os.path.join(self.path.eval_func, "model"),
-            save_residuals=os.path.join(self.path.eval_func, "residuals.txt")
+            save_residuals=os.path.join(self.path.eval_func,
```

Comment:
here also I think we can explicitly name the iteration and step count. This would also be useful for reading back in the residual files because then we know we're using the same values. e.g,.

```python
 def _evaluate_line_search_misfit(self):
        """Convenience fuinction to wrap forward solver and misfit calc"""
        # Define where we are in the inversion for file passing between
        # preprocess and workflow modules
        iteration = self.iteration
        step_count = self.optimize.step_count + 1

        self.system.run(
            [self.run_forward_simulations,
             self.evaluate_objective_function],
            path_model=os.path.join(self.path.eval_func, "model"),
            save_residuals=os.path.join(self.path.eval_func,
                                        f"residuals_{{src}}_{iteration}_{step_count}.txt")
        )

        residuals_files = glob(os.path.join(self.path.eval_func,
                                            f"residuals_*_{iteration}_{step_count}.txt"))

        residuals = self.preprocess.read_residuals(residuals_files)
```